packages/ui-kit/src/components/multi-code-block.js

```diff
@@ -88,11 +88,34 @@ const languageDisplayNames = {
 function MultiCodeBlock(props) {
   const langs = extractLanguages(props.children);
 
-  const [selected, setSelected] = React.useState(langs[0]);
+  const [selected, setSelected] = React.useState(
+    langs.find(
+      (lang) =>
+        // connfirm if selectedLanguage exists in list of languages
+        lang === props.selectedLanguage
+    ) || langs[0]
+  );
+
+  React.useEffect(() => {
+    // do only if this has multile code examples
+    if (langs.length > 1) {
+      const langIndex = langs.findIndex(
+        (lang) => lang === props.selectedLanguage
+      );
+
+      // will set selected language if
+      // - the language exists for this instance
+      // - this is not the first render.
+      // - this is not the instance that set the language
+      if (langIndex > -1 && selected !== props.selectedLanguage) {
+        setSelected(props.selectedLanguage);
+      }
+    }
+  }, [props.selectedLanguage, langs, selected]);
```

Comment:
Hmm I feel like this code-example <-> context setup is odd.

Ideally, the context should be initialized with the "selected" language. Then the `MultiCodeBlock` component simply reads from the context and updates the value only when the user selects the language.

Here we're trying to "shoot ourselves in the foot" by potentially running into an infinite loop, if we're not careful with state and the effect.

I'll have to think about it if there is a more safer approach.

Reply:
(_talking out lout for myself_)

However, because of SSR, we can't know the value of the selected language (from localstorage) when the static pages are created.
So by default, the first language in the list will be always be rendered.
Then, the context gets initialized with the localstorage value.
At this point the component re-renders and the language context may contain a value. If the value is defined, the selected language should reflect the value from the context, but only if the language is available within the rendered code examples.

This is kind of tricky.

For instance, let's say we have code examples in 3 languages: java, javascript, php.
I would assume that ALL multi code examples should contain examples for each language, otherwise it would be weird if all examples on a page switch to php but some of them don't have an associated example file.
What should happen in this case? Should we fall back to one of the other examples or should we render a "blank" example?

Then maybe we need to fix the "multi code examples setup" first. For example, we need to specify which languages are supposed to be rendered, and that should be the same for all of them.

cc @nkuehn ?